samza-core/src/main/java/org/apache/samza/processor/StreamProcessor.java

```diff
@@ -359,8 +362,12 @@ public void onNewJobModel(String processorId, JobModel jobModel) {
       public void onCoordinatorStop() {
         synchronized (lock) {
           LOGGER.info("Shutting down the executor service of the stream processor: {}.", processorId);
-          stopSamzaContainer();
-          executorService.shutdownNow();
+          boolean hasContainerShutdown = stopSamzaContainer();
+
+          // we only want to interrupt when container shutdown times out.
```

Comment:
Why? Is something else running on this executor other than the container?
If not, would also prefer renaming this to containerExecutorService to clarify it's purpose.

Reply:
> Is something else running on this executor other than the container

There will be nothing running other than SamzaContainer within that executor service. Renaming should be  fine.